src/vs/vscode.proposed.d.ts

```diff
@@ -881,6 +881,28 @@ declare module 'vscode' {
 	}
 	//#endregion
 
+	//#region Custom Tree View Drag and Drop https://github.com/microsoft/vscode/issues/32592
+	export interface TreeViewOptions<T> {
+		/**
+		 * * Whether the tree supports drag and drop.
+		 */
+		enableDragAndDrop?: boolean;
```

Comment:
To align more closely with the existing `TreeViewOptions`, let's call this `canDragAndDrop`.